src/image.rs

```diff
@@ -133,6 +133,77 @@ impl ImageFormat {
         inner(path.as_ref())
     }
 
+    /// Return if the ImageFormat can be decoded by the lib.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use image::ImageFormat;
+    /// assert_eq!(ImageFormat::Png.can_read(), true);
+    /// ```
+    #[inline]
+    pub fn can_read(&self) -> bool {
+        // Needs to be updated once a new variant's decoder is added to free_functions.rs::load
+        match self {
+            #[cfg(feature = "png")]
+            ImageFormat::Png => true,
+            #[cfg(feature = "gif")]
+            ImageFormat::Gif => true,
+            #[cfg(feature = "jpeg")]
+            ImageFormat::Jpeg => true,
+            #[cfg(feature = "webp")]
+            ImageFormat::WebP => true,
+            #[cfg(feature = "tiff")]
+            ImageFormat::Tiff => true,
+            #[cfg(feature = "tga")]
+            ImageFormat::Tga => true,
+            #[cfg(feature = "dds")]
+            ImageFormat::Dds => true,
+            #[cfg(feature = "bmp")]
+            ImageFormat::Bmp => true,
+            #[cfg(feature = "ico")]
+            ImageFormat::Ico => true,
+            #[cfg(feature = "hdr")]
+            ImageFormat::Hdr => true,
+            #[cfg(feature = "pnm")]
+            ImageFormat::Pnm => true,
+            #[cfg(feature = "farbfeld")]
+            ImageFormat::Farbfeld => true,
+            _ => false,
+        }
+    }
+
+    /// Return if the ImageFormat can be encoded by the lib.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use image::ImageFormat;
+    /// assert_eq!(ImageFormat::Png.can_write(), true);
+    /// assert_eq!(ImageFormat::Pnm.can_write(), false);
+    /// ```
+    #[inline]
+    pub fn can_write(&self) -> bool {
+        // Needs to be updated once a new variant's encoder is added to free_functions.rs::save_buffer_with_format_impl
```

Comment:
I noticed that `free_functions.rs::save_buffer_with_format_impl` & `free_functions.rs::save_buffer_impl` have different match clauses as in `free_functions.rs::save_buffer_with_format_impl` has no references to `PnmEncoder` is that inteded? Can I fix that in a follow up PR?

Reply:
Or should I just implement what https://github.com/image-rs/image#supported-image-formats says we support?

Reply:
I'd keep it simple for now. The difference in the two methods seems to be indeed an error.

Reply:
Opened #1390 for it.